src/Microsoft.Data/ArrowStringColumn.cs

```diff
@@ -0,0 +1,348 @@
+            ArrowBuffer dataBuffer = numberOfRows == 0 ? ArrowBuffer.Empty : new ArrowBuffer(_dataBuffers[offsetsBufferIndex].ReadOnlyMemory);
+            ArrowBuffer offsetsBuffer = numberOfRows == 0 ? ArrowBuffer.Empty : new ArrowBuffer(_offsetsBuffers[offsetsBufferIndex].ReadOnlyMemory);
+            ArrowBuffer nullBuffer = numberOfRows == 0 ? ArrowBuffer.Empty : new ArrowBuffer(_nullBitMapBuffers[offsetsBufferIndex].ReadOnlyMemory);
+            int nullCount = GetNullCount(startIndex, numberOfRows);
+            return new StringArray(numberOfRows, offsetsBuffer, dataBuffer, nullBuffer, nullCount, (int)startIndex);
+        }
+
+        public override BaseColumn Sort(bool ascending = true) => throw new NotSupportedException();
+
+        public override BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapIndices = false, long numberOfNullsToAppend = 0)
+        {
+            ArrowStringColumn clone;
+            if (!(mapIndices is null))
+            {
+                if (mapIndices.DataType != typeof(long))
+                    throw new ArgumentException(Strings.MismatchedValueType + " PrimitiveColumn<long>", nameof(mapIndices));
+                if (mapIndices.Length > Length)
+                    throw new ArgumentException(Strings.MapIndicesExceedsColumnLenth, nameof(mapIndices));
+                clone = Clone(mapIndices as PrimitiveColumn<long>, invertMapIndices);
+            }
+            else
+            {
+                clone = Clone();
+            }
+            for (long i = 0; i < numberOfNullsToAppend; i++)
+            {
+                clone.Append(null);
+            }
+            return clone;
+        }
+
+        private ArrowStringColumn Clone(PrimitiveColumn<long> mapIndices = null, bool invertMapIndex = false)
+        {
+            ArrowStringColumn ret = new ArrowStringColumn(Name);
+            if (mapIndices is null)
+            {
+                for (long i = 0; i < Length; i++)
+                {
+                    ret.Append(this[i]);
+                }
+            }
+            else
+            {
+                if (mapIndices.Length > Length)
+                    throw new ArgumentException(Strings.MapIndicesExceedsColumnLenth, nameof(mapIndices));
+                if (invertMapIndex == false)
+                {
+                    for (long i = 0; i < mapIndices.Length; i++)
+                    {
+                        ret.Append(this[(long)mapIndices[i]]);
+                    }
+                }
+                else
+                {
+                    long mapIndicesLengthIndex = mapIndices.Length - 1;
+                    for (long i = 0; i < mapIndices.Length; i++)
+                    {
+                        ret.Append(this[(long)mapIndices[mapIndicesLengthIndex - i]]);
+                    }
+                }
+            }
+            return ret;
+        }
+
+        public Dictionary<TKey, ICollection<long>> HashColumnValues<TKey>()
```

Comment:
Who calls this method?

Reply:
Don't you want to override `GroupColumnValues` instead?

---
In reply to: [307387981](https://github.com/dotnet/corefxlab/pull/2701#discussion_r307387981) [](ancestors = 307387981)

Reply:
> Don't you want to override GroupColumnValues instead?

Yup. I missed renaming this somehow. Fixed